browser/src/layer/vector/SVGGroup.js

```diff
@@ -78,6 +80,62 @@ L.SVGGroup = L.Layer.extend({
 		var point = this._map.latLngToLayerPoint(this._bounds.getNorthWest());
 		svgLastChild.setAttribute('x', point.x);
 		svgLastChild.setAttribute('y', point.y);
+
+		var video = svgLastChild.getElementsByTagName('video')[0];
+		this.checkVideoSupport(video);
+	},
+
+	checkVideoSupport: function(video) {
+		this.video_ = video;
+		this.startTime_ = 0;
+		this.decodedFrames_ = 0;
+		this._videoInterval = null;
+		var that = this;
+
+		this.video_.addEventListener('playing', function() {
+			that.onVideoPlaying();
+		});
+
+		this.video_.addEventListener('error', function() {
+			that.endVideoSupportCheck();
+			this._map.uiManager.showSnackbar(_('Document contains uncompatible video'));
+		});
+
+		this.video_.addEventListener('ended', function() {
+			that.endVideoSupportCheck();
+		});
+	},
+
+	endVideoSupportCheck: function() {
+		window.clearInterval(this._videoInterval);
+	},
+
+	onVideoPlaying: function() {
+		this.decodedFrames_ = 0;
+		this.startTime_ = window.performance.now();
+		this.endVideoSupportCheck();
+		var that = this;
+		this._videoInterval = window.setInterval(function() {
+			that.calculateFrames();}, 1000);
+	},
+
+	calculateFrames: function() {
+		if (this.video_.readyState <= HTMLMediaElement.HAVE_CURRENT_DATA ||
+			this.video_.paused || this.video_.ended)
+		  return;
+		var currentTime = window.performance.now();
+		var deltaTime = (currentTime - this.startTime_) / 1000;
+		this.startTime_ = currentTime;
+
+		// Calculate decoded frames per sec.
+		var fps = (this.video_.webkitDecodedFrameCount - this.decodedFrames_) /
+				   deltaTime;
+		this.decodedFrames_ = this.video_.webkitDecodedFrameCount;
```

Comment:
Isn't it enough to check that `webkitDecodedFrameCount` is positive? Why do we care about the frame rate at all. If after a reasonable time (500ms or 1s) we don't have any frames, there probably is no video being played back.

This should work for videos, but I don't expect it to work for audio-only. Perhaps that's ok (but we shouldn't complain that an audio isn't supported when it is playing loud and clear, so best to ignore audio-only media).

Reply:
good point, indeed we don't need fps :) I'll take care of it